### PR TITLE
Tf2 dev

### PR DESCRIPTION
**Pull request by jonathanbratt**
still a bit to clean up, specifically in the "functions to improve" file, but this update let me take out almost all the tokenization stuff!

There's still a kinda fuzzy line between wordpiece and RBERT, and to what extent we hard-code BERT conventions in. So some stuff might still need to get moved between the packages.

All the tests should pass when run directly, but there's still one of the tests (downloading checkpoint, I think) that fails in `gp()`. I think it's been like that for a while; haven't dug into it lately.

### CODE REVIEW

File: R/tokenization.R [hunk 1]
```diff
@@ -13,768 +13,140 @@
...
 #'
-#' @return an object of class BasicTokenizer
+#' @return The tokenized output as a named integer vector, with token type ids
+#'   attached as an attribute.
 #' @export
-#'
 #' @examples
 #' \dontrun{
-#' b_tokenizer <- BasicTokenizer(TRUE)
+#' # assuming vocabulary `vocab`
```
Comment (jonthegeek) [inline R/tokenization.R:45]: Can we update this to use the built-in {wordpiece} vocab, so it doesn't have to be dontrun?

File: tests/testthat/test-tokenization.R [hunk 2]
```diff
@@ -12,118 +12,4 @@
...
-  # tests from BERT: tokenization_test.py
-  testthat::expect_true(is_punctuation("-"))
-  testthat::expect_true(is_punctuation("$"))
-  testthat::expect_true(is_punctuation("`"))
-  testthat::expect_true(is_punctuation("."))
-
-  testthat::expect_false(is_punctuation("A"))
-  testthat::expect_false(is_punctuation(" "))
-})
+# maybe new tests here?
```
Comment (jonthegeek) [inline tests/testthat/test-tokenization.R:15]: I'd say either add relevant tests here or delete this file (but maybe add a ticket either way).

**Review by jonthegeek — CHANGES_REQUESTED** [review]
Some changes, some prompts for discussions.

File: R/tf2-functions-to-improve.R [hunk 3]
```diff
@@ -50,23 +51,19 @@ run_bert_model_on_text <- function(bert_model, text) {
...
-    tmp <- convert_tokens_to_ids(vocab = vocab, tokens = tt)
-    c(tmp, rep(0, n_token_max-length(tmp))) # pad with zeros
-  })
-  #TODO add check for n_token_max here.
-  # Update this to work with multi-segment inputs.
-  ttypes <- purrr::map(tokenized_text, function(tt) {
-    rep(0, n_token_max)
-  })
+  vocab <- wordpiece::load_or_retrieve_vocab(vocab_file = vocab_file,
+                                             use_cache = TRUE)
```
[the remark this answers is not in the document]
Comment (jonathanbratt) [reply R/tf2-functions-to-improve.R:60]: yeah, probably.

**Comment by jonathanbratt** [comment]
also, this PR closes #55

File: tests/testthat/test-tokenization.R (hunk 2 above)
Comment (jonathanbratt) [reply tests/testthat/test-tokenization.R]: deleted file.

**Review by jonthegeek — APPROVED** [review]
Looks great!

**Pull request merged**